internal/fwschemadata/data_nullify_collection_blocks.go

```diff
@@ -0,0 +1,81 @@
+package fwschemadata
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/internal/fromtftypes"
+	"github.com/hashicorp/terraform-plugin-framework/internal/fwschema"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+// NullifyCollectionBlocks returns a copy of the Data where list and set block
+// empty values are converted to null values.
+func (d Data) NullifyCollectionBlocks(ctx context.Context) (Data, diag.Diagnostics) {
```

Comment:
Believe that `Data` has methods on both value and pointer receivers.

Reply:
Updated to a pointer receiver method and removed the copying/return.